tests/e2e/specs/stories-editor/media-file-size-warnings.test.js

```diff
@@ -0,0 +1,99 @@
+
+/**
+ * Tests the notices for excessive video size.
+ *
+ * These are similar to the tests in video-poster-image-extraction.test.js.
+ */
+describe( 'Media File Size Warnings', () => {
+	beforeAll( async () => {
+		await activateExperience( 'stories' );
+	} );
+
+	afterAll( async () => {
+		await deactivateExperience( 'stories' );
+	} );
+
+	describe( 'Background Media', () => {
+		it( 'should display a warning in the Media Library when a large video is uploaded', async () => {
+			await createNewPost( { postType: 'amp_story' } );
+
+			// Select the default page block.
+			await selectBlockByClientId(
+				( await getAllBlocks() )[ 0 ].clientId
+			);
+
+			// Click the media selection button.
+			await page.waitForSelector( '.editor-amp-story-page-background' );
+			await page.click( '.editor-amp-story-page-background' );
+			await uploadMedia( OVERSIZED_VIDEO );
+
+			// The warning Notice component should appear.
+			await expect( page ).toMatchElement( '.media-toolbar-secondary .notice-warning' );
+
+			// The warning notice text should appear.
+			await expect( page ).toMatch( EXPECTED_NOTICE_TEXT );
+
+			// It should be possible to insert the uploaded video, as this should not disable the 'Select' button.
+			await expect( page ).toClick( SELECT_BUTTON );
+		} );
+
+		it( 'should not display a warning in the Media Library for an appropriate video file size', async () => {
+			await createNewPost( { postType: 'amp_story' } );
+
+			// Select the default page block.
+			await selectBlockByClientId(
+				( await getAllBlocks() )[ 0 ].clientId
+			);
+
+			// Click the media selection button.
+			await page.waitForSelector( '.editor-amp-story-page-background' );
+			await page.click( '.editor-amp-story-page-background' );
+			await uploadMedia( 'clothes-hanged-to-dry-1295231.mp4' );
+
+			// The warning Notice component should not appear.
+			await expect( page ).not.toMatchElement( '.media-toolbar-secondary .notice-warning' );
+
+			// The warning notice text should not appear.
+			await expect( page ).not.toMatch( EXPECTED_NOTICE_TEXT );
+
+			// It should be possible to insert the uploaded video.
+			await expect( page ).toClick( SELECT_BUTTON );
+		} );
+	} );
+
+	describe( 'Video Block', () => {
+		it( 'should display a warning in the Media Library when a large video is uploaded', async () => {
+			await createNewPost( { postType: 'amp_story' } );
+
+			// Using the regular inserter.
+			await insertBlock( 'Video' );
+
+			// Click the media library button.
+			await page.waitForSelector( '.editor-media-placeholder__media-library-button' );
+			await page.click( '.editor-media-placeholder__media-library-button' );
+			await uploadMedia( OVERSIZED_VIDEO );
+
+			// The warning Notice component should appear.
+			await expect( page ).toMatchElement( '.media-toolbar-secondary .notice-warning' );
+
+			// The warning notice text should appear.
+			await expect( page ).toMatch( EXPECTED_NOTICE_TEXT );
```

Comment:
This checks whether the text exists anywhere on the page, not necessarily in the media modal.

This seems more explicit:

```js
const warningNotice = await page.$( '.media-toolbar-secondary .notice-warning' );
await expect( warningNotice ).toMatch( EXPECTED_NOTICE_TEXT );
```

Reply:
Yeah, good point. I'll change that.

Reply:
Thanks, https://github.com/ampproject/amp-wp/pull/2927/commits/b9d00d11c16d6577d495ed5a97349d7c86165a20 uses your snippet above. 

Also, it enables removing:

```js
await expect( page ).toMatchElement( '.media-toolbar-secondary .notice-warning' );
```

...as it's going to await the element anyway, 3 lines of code below:

```js
const warningNotice = await page.$( NOTICE_SELECTOR );
```

Reply:
Excellent